include/grass/defs/gis.h

```diff
@@ -545,6 +545,9 @@ int G_name_is_fully_qualified(const char *, char *, char *);
 char *G_fully_qualified_name(const char *, const char *);
 int G_unqualified_name(const char *, const char *, char *, char *);
 
+/* omp_threads.c */
+int G_setup_threads(char *);
```

Comment:
I would directly pass `const struct Option *option` instead of a string.

Reply:
The standard option is called `NPROCS` and this function is `G_setup_threads()`. I initially thought it would be great to have consistent naming (`G_setup_nprocs()`). Then, `NPROCS` can mean three different things in parallelization: (1) number of cores, (2) number of threads, and (3) number of GPU threads. However, the option itself is agnostic of these concepts. Let's say, we support MPI or GPU later, then we can still use the same `NPROCS` for those (or new options for hybrid parallelism), and we can add `G_setup_processes()` and `G_setup_GPU_threads()`, for example.

Reply:
Should we use `G_setup_omp_threads` to be more specific, or do you think `G_setup_threads` is enough?

Reply:
I think we should use the more specific `G_setup_omp_threads()` because that's exactly what we do. We don't set up other threading methods here (e.g., POSIX threads). https://www.cs.colostate.edu/~cs675/OpenMPvsThreads.pdf

Reply:
Also, I just feel like this function *set* the OMP number of threads (instructing to use this number of OMP threads), not *set* it *up* (preparing OpenMP threads), so how about `G_set_omp_num_threads()`?